src/pages/ReportDetailsPage.js

```diff
@@ -56,146 +56,142 @@ const defaultProps = {
+    const isThread = useMemo(() => ReportUtils.isThread(props.report), [props.report]);
+    const chatRoomSubtitle = useMemo(() => ReportUtils.getChatRoomSubtitle(props.report), [props.report]);
 
-    getMenuItems() {
-        const menuItems = [
+    const menuItems = useMemo(() => {
+        const items = [
             {
                 key: CONST.REPORT_DETAILS_MENU_ITEM.SHARE_CODE,
                 translationKey: 'common.shareCode',
                 icon: Expensicons.QrCode,
-                action: () => Navigation.navigate(ROUTES.getReportShareCodeRoute(this.props.report.reportID)),
+                action: () => Navigation.navigate(ROUTES.getReportShareCodeRoute(props.report.reportID)),
             },
         ];
 
-        if (ReportUtils.isArchivedRoom(this.props.report)) {
+        if (ReportUtils.isArchivedRoom(props.report)) {
             return [];
         }
 
-        if (lodashGet(this.props.report, 'participants', []).length) {
-            menuItems.push({
+        if (lodashGet(props.report, 'participants', []).length) {
+            items.push({
                 key: CONST.REPORT_DETAILS_MENU_ITEM.MEMBERS,
                 translationKey: 'common.members',
                 icon: Expensicons.Users,
-                subtitle: lodashGet(this.props.report, 'participants', []).length,
+                subtitle: lodashGet(props.report, 'participants', []).length,
                 action: () => {
-                    Navigation.navigate(ROUTES.getReportParticipantsRoute(this.props.report.reportID));
+                    Navigation.navigate(ROUTES.getReportParticipantsRoute(props.report.reportID));
                 },
             });
         }
 
-        if (ReportUtils.isPolicyExpenseChat(this.props.report) || ReportUtils.isChatRoom(this.props.report) || ReportUtils.isThread(this.props.report)) {
-            menuItems.push({
+        if (ReportUtils.isPolicyExpenseChat(props.report) || ReportUtils.isChatRoom(props.report) || ReportUtils.isThread(props.report)) {
+            items.push({
                 key: CONST.REPORT_DETAILS_MENU_ITEM.SETTINGS,
                 translationKey: 'common.settings',
                 icon: Expensicons.Gear,
                 action: () => {
-                    Navigation.navigate(ROUTES.getReportSettingsRoute(this.props.report.reportID));
+                    Navigation.navigate(ROUTES.getReportSettingsRoute(props.report.reportID));
                 },
             });
         }
 
-        const policy = this.getPolicy();
-        const isThread = ReportUtils.isThread(this.props.report);
-        if (ReportUtils.isUserCreatedPolicyRoom(this.props.report) || ReportUtils.canLeaveRoom(this.props.report, !_.isEmpty(policy)) || isThread) {
-            menuItems.push({
+        if (ReportUtils.isUserCreatedPolicyRoom(props.report) || ReportUtils.canLeaveRoom(props.report, !_.isEmpty(policy)) || isThread) {
```

Comment:
Looks like we can use the variables defined at the top in these ifs, judging by the use of `isThread` here, if that's the case, would it make sense to create variables at the top to hold the values of `ReportUtils.isUserCreatedPolicyRoom(props.report)` and `ReportUtils.canLeaveRoom(props.report, !_.isEmpty(policy))`?

Reply:
Sure, we can then remove the dependency on other policy fields and only leave `props.report.reportID` as a report dependency :)